api/src/main/java/au/org/aodn/nrmn/restapi/service/TemplateService.java

```diff
@@ -173,8 +170,12 @@ public void writeSpeciesCsv(Writer writer, CSVFormat csvFormat, Collection<Speci
                 toString(species.getLMax()));
     }
 
-    public List<SpeciesWithAttributesCsvRow> getSpeciesForTemplate(Integer mode, Collection<Integer> siteIds,
-            HashMap<Long, String> letterCodeMap) {
+    public List<SpeciesWithAttributesCsvRow> getSpeciesForTemplate(Integer mode, List<Integer> siteIds) {
+
+        HashMap<Long, String> letterCodeMap = new HashMap<Long, String>();
```

Comment:
You do not need to write the type here, which is less verbose.

HashMap<Long, String> letterCodeMap = new HashMap<>();